wgpu/src/backend/direct.rs

```diff
@@ -163,6 +163,21 @@ impl Context {
         }
     }
 
+    #[cfg(all(target_arch = "wasm32", feature = "webgl"))]
+    pub unsafe fn create_surface_from_canvas(
+        self: &Arc<Self>,
+        canvas: &web_sys::HtmlCanvasElement,
+    ) -> crate::Surface {
+        let id = self.0.create_surface_webgl_canvas(canvas, PhantomData);
```

Comment:
@zicklag Your suggestions look promising. I've started with `create_surface_from_canvas()` because it should be easier to accomplish. I've managed to get a first version, except on compile issue I didn't manage to solve.

When building with `cargo build --target wasm32-unknown-unknown --features webgl` I get the following error:
```
error[E0599]: no method named `create_surface_webgl_canvas` found for struct `wgpu_core::hub::Global` in the current scope
   --> wgpu\src\backend\direct.rs:171:25
    |
171 |         let id = self.0.create_surface_webgl_canvas(canvas, PhantomData);
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `wgpu_core::hub::Global<IdentityManagerFactory>`
```

It's unclear to me, how `self.0` is connected to `instance` from wasm-core. The type is the tuple-struct [`pub struct Context(wgc::hub::Global<wgc::hub::IdentityManagerFactory>)`](https://github.com/gfx-rs/wgpu/blob/9743068b4335762d82ed8c784678b8b599e559fd/wgpu/src/backend/direct.rs#L26), but this trail leaves more questions than answers for me 😜  It looks like the same code works for metals `instance_create_surface_metal()`.

Reply:
Hey @haraldreingruber-dedalus, I'm glad to help! :)

I just pushed a commit that fixes compilation, but I didn't test anything, so you'll have to try it out!

https://github.com/gfx-rs/wgpu/commit/7696677a538980a22e4cc870e47a93bf4f2c9cb7

The issue was the `#[cfg]` check above the `create_surface_webgl_canvas` function looked for a `webgl` feature, but the webgl feature doesn't exist on the `wgpu-core` crate!

The other tiny fix necessary was to add an optional dependency on `web-sys` for the `wgpu-core` crate.'

I can't remember if, when I was working on the WebGL backend, there was some reason we wanted to keep the `webgl` feature and `web-sys` dependency out of the `wgpu-core` create, or not, though, so that will be something to ask for review on.

In this scenario I think it makes sense, but I just can't remember if there was a reason to try and avoid it or not.

Reply:
Awesome, thanks for catching this!